### ISSUE TITLE
The value of disparity is extreme small by using the model trained on my stereo dataset

### ISSUE THREAD

**Issue by TommyAnqi** [mrharicot/monodepth#75]
First, Thanks for your amazing work.
I have used 4500 stereo samples collected from a simulated platform to train the model. The image size is 780*560. But the value of disparity is very small (2.80002571e-36.. like this ) by using this model.
Do you have any idea about it?

**Comment by mrharicot** [comment]
I was expecting something like 0.01 but no!
2.80002571e-36 is indeed amazingly small!
Can you share the details of how you trained the model? And maybe a couple of images that you use for training?

**Comment by TommyAnqi** [comment]
I just follow the training steps introducing in ReadMe and didn't change anything except the data path. The model was trained on 4500 samples collected from GTA V. Two examples are attached. 

Actually, I have tested a single GTA V image by pre-trained model, the result is not bad. 
I wonder to train the model on GTA V data and try to get a better result. But the result of disparity shows most of the value are 0 and several values are extreme small as I have mentioned.....

I am trying to solve this problem by editing the stereo model, as there is no stereo model in GTA V and I made a script to generate a second camera. So, I guess maybe the problem is the camera making the input images inappropriate...but still not sure. 



![cameraleft_75943](https://user-images.githubusercontent.com/24237343/33072044-dd77fc48-cef8-11e7-8e03-b96ba099f2ec.jpg)
![cameraright_75943](https://user-images.githubusercontent.com/24237343/33072046-dddc810e-cef8-11e7-8c09-37bc769f2531.jpg)
![cameraleft_76515](https://user-images.githubusercontent.com/24237343/33072045-ddaa649e-cef8-11e7-94fe-b9bd0f9bdf80.jpg)
![cameraright_76515](https://user-images.githubusercontent.com/24237343/33072047-de0dd54c-cef8-11e7-900f-34a408becb7e.jpg)

**Comment by mrharicot** [comment]
The stereo pairs seem to be fine, everything moves horizontally when you switch back and forth.
A few minor issues: 
-the pedestrians seem to keep moving between each image, while everything else is static
-texture filtering is quite low: try using anisotropic filtering x8 or x16 to avoid blurring the road texture
-aliasing issues: use either a higher resolution, MSAA or both.

Do you use the default 256x512 size when training?

How varied are your 4.5k images? How often do you capture a stereo pair while driving?

Finally, did you try training from the pretrained cityscapes model instead of training from scratch?

**Comment by TommyAnqi** [comment]
yeah...the script only freeze the position of pedestrians but they are not really static. I am trying to find another function to replace it.
ok, I will improve the image quality and retrain the model.
Thanks for your suggestions.

The input image size is 560*780, and the output result is 256*512. 

All the images are in sunny condition. 60% of the images are collected in the city scene, 20% in the highway and the rest are coming from the village or town lane. 3 seconds one stereo sample. Do you think I need to increase the capturing frequency? 

The results from the pre-trained cityscapes model seem better than that from Kitti model.

**Comment by mrharicot** [comment]
Do you train from scratch or from one of the pretrained models?
Do you shuffle the filenames before training?
How is the camera attached to the car? It seems to move quite a bit between those two scenes.

**Comment by TommyAnqi** [comment]
I trained the model from scratch.
Oh.....I didn't  shuffle the filenames.

I created two cameras attaching on the front of the car, the distance between them is 0.2m. And the field of view is 30 degree.

I have collected 20k new stereo images with higher quality and the distance of two cameras are adjusted to 0.6m. I will shuffle the filenames and see the training results....

**Comment by TommyAnqi** [comment]
@mrharicot  I have finished training a model by only using the GTAV stereo images. The results look pretty good. I will try more model combinations. Thanks a lot.
![cameraleft_69459](https://user-images.githubusercontent.com/24237343/33248794-fae09b10-d361-11e7-8912-00015b0bb6cc.jpg)
![cameraleft_69459_disp](https://user-images.githubusercontent.com/24237343/33248795-fb19eb7c-d361-11e7-98a3-d50345f526e6.png)

**Comment by mrharicot** [comment]
Really glad you got it to work!
Final question, what format are you storing your images in?
The images here are quite compressed, I would suggest using .png at first and then convert to .jpeg using imagemagick:
`mogrify -format jpg *.png`
By default the compression rate is chosen so that there is not visible difference from the original image. But it reduces the file size significantly.

**Comment by TommyAnqi** [comment]
Thanks for your suggestion, I used the python default conversion. I will try to use imagemagick.
Thank you for your great work again.

**Comment by Sarah20187** [comment]
> I just follow the training steps introducing in ReadMe and didn't change anything except the data path. The model was trained on 4500 samples collected from GTA V. Two examples are attached.
> 
> Actually, I have tested a single GTA V image by pre-trained model, the result is not bad. I wonder to train the model on GTA V data and try to get a better result. But the result of disparity shows most of the value are 0 and several values are extreme small as I have mentioned.....
> 
> I am trying to solve this problem by editing the stereo model, as there is no stereo model in GTA V and I made a script to generate a second camera. So, I guess maybe the problem is the camera making the input images inappropriate...but still not sure.
> 
> ![cameraleft_75943](https://user-images.githubusercontent.com/24237343/33072044-dd77fc48-cef8-11e7-8e03-b96ba099f2ec.jpg) ![cameraright_75943](https://user-images.githubusercontent.com/24237343/33072046-dddc810e-cef8-11e7-8c09-37bc769f2531.jpg) ![cameraleft_76515](https://user-images.githubusercontent.com/24237343/33072045-ddaa649e-cef8-11e7-94fe-b9bd0f9bdf80.jpg) ![cameraright_76515](https://user-images.githubusercontent.com/24237343/33072047-de0dd54c-cef8-11e7-900f-34a408becb7e.jpg)

hi there! I also want to generate stereo pairs from GTA V dataset. Could you tell me how to do that? Thanks a lot!
